### DISCUSSION TITLE
Different results locally(laptop, say 32 threads) and on the server(128 threads)

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD#5273] [General]
Q: what happens if we have different number of cores? Should we expect different results after `make place` even if we use manual placement?

I suppose OpenROAD can use all available CPUs and not heed NUM_CORES in ORFS, so even if I set NUM_CORES to the same, that might not help me get the exact same results?

I would expect partitioning to affect the results, is this true?

If so, can I choose the same partitioning even if NUM_CORES differ?

Any thoughts on different results locally and on the server in terms of what is expected behavior?

**Comment by QuantamHD** [comment]
It might be this issue https://github.com/The-OpenROAD-Project/OpenROAD/pull/5234

**Reply by oharboe** [reply]
I see.... Can you comment on my original question as well: what should I expect?

**Reply by oharboe** [reply]
What is the intended and desired behaviour?

What might I realistically expect of the current code?

**Reply by maliberty** [reply]
You should expect the same result regardless of number of threads.

**Reply by oharboe** [reply]
I see. In that case, I need to create a testcase for why I get different results....

**Comment by oharboe** [comment]
For now my assumption is a PEBCAK: that I am running two different versions of OpenROAD in CI and locally(multiple levels of docker hell).

I tried locally with a single level of docker and got the same results on two machines and varying NUM_CORES.

**Comment by maliberty** [comment]
fwiw https://github.com/The-OpenROAD-Project/OpenROAD/pull/5234 is merged
